mlflow/store/artifact/databricks_model_artifact_repo.py

```diff
@@ -0,0 +1,144 @@
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        super().__init__(artifact_uri)
+        from mlflow.tracking import MlflowClient
+
+        databricks_profile_uri = (
+            get_databricks_profile_uri_from_artifact_uri(artifact_uri) or mlflow.get_registry_uri()
+        )
+
+        self.client = MlflowClient(registry_uri=databricks_profile_uri)
+        hostcreds_from_uri = get_databricks_host_creds(databricks_profile_uri)
+        self.get_host_creds = lambda: hostcreds_from_uri
+        self.model_name, self.model_version = self._get_model_name_and_version(artifact_uri)
+
+    def _get_model_name_and_version(self, artifact_uri):
+        (model_name, model_version, model_stage) = parse_model_uri(artifact_uri)
+        if model_stage is not None:
+            model_version = get_model_version_from_stage(self.client, model_name, model_stage)
+        return model_name, str(model_version)
+
+    def _call_endpoint(self, json, endpoint):
+        host_creds = self.get_host_creds()
+        return http_request(host_creds=host_creds, endpoint=endpoint, method="GET", params=json)
+
+    def _make_json_body(self, path, page_token=None):
+        if page_token:
+            return {"name": self.model_name, "version": self.model_version, "path": path}
+        return {
+            "name": self.model_name,
+            "version": self.model_version,
+            "path": path,
+            "page_token": page_token,
+        }
+
+    def list_artifacts(self, path=None):
+        infos = []
+        page_token = None
+        if not path:
+            path = ""
+        while True:
+            json_body = self._make_json_body(path, page_token)
+            response = self._call_endpoint(json_body, REGISTRY_LIST_ENDPOINT)
+            try:
+                json_response = json.loads(response.text)
+            except ValueError:
+                raise MlflowException(
+                    "API request to list files under path `%s` failed with status code %s. "
+                    "Response body: %s" % (path, response.status_code, response.text)
+                )
+            artifact_list = json_response.get("files", [])
+            next_page_token = json_response.get("next_page_token", None)
+            # If `path` is a file, ListArtifacts returns a single list element with the
+            # same name as `path`. The list_artifacts API expects us to return an empty list in this
+            # case, so we do so here.
+            if (
+                len(artifact_list) == 1
+                and artifact_list[0]["path"] == path
+                and not artifact_list[0]["is_dir"]
+            ):
+                return []
+            for output_file in artifact_list:
+                artifact_size = None if output_file["is_dir"] else output_file["file_size"]
+                infos.append(FileInfo(output_file["path"], output_file["is_dir"], artifact_size))
+            if len(artifact_list) == 0 or not next_page_token:
```

Comment:
So currently it will not happen, since the server does not return a page token at all (there is no pagination on the backend).
However, this condition should never arise and I am sure we will have tests and check guarding against this edge case.